google_bigquery_writer/writer.py

```diff
@@ -121,12 +121,19 @@ def write_table(
             job_config.skip_leading_rows = 1
             job_config.allow_quoted_newlines = True
 
-            job = self.bigquery_client.load_table_from_file(
-                readable,
-                table_reference,
-                job_config=job_config
-            )
-
+            try:
+                job = self.bigquery_client.load_table_from_file(
+                    readable,
+                    table_reference,
+                    job_config=job_config
+                )
+            except bq_exceptions.BadRequest as err:
```

Comment:
Jo, tak ještě bacha! Tohle je stacktrace z té issue v Jíře

```
ConnectionResetError(104, 'Connection reset by peer'))
Traceback (most recent call last):
  File \"/usr/local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 672, in urlopen
    chunked=chunked,
  File \"/usr/local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 421, in _make_request
    six.raise_from(e, None)
  File \"<string>\", line 3, in raise_from
  File \"/usr/local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 416, in _make_request
    httplib_response = conn.getresponse()
  File \"/usr/local/lib/python3.6/http/client.py\", line 1354, in getresponse
    response.begin()
  File \"/usr/local/lib/python3.6/http/client.py\", line 307, in begin
    version, status, reason = self._read_status()
  File \"/usr/local/lib/python3.6/http/client.py\", line 268, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")
  File \"/usr/local/lib/python3.6/socket.py\", line 586, in readinto
    return self._sock.recv_into(b)
  File \"/usr/local/lib/python3.6/ssl.py\", line 1012, in recv_into
    return self.read(nbytes, buffer)
  File \"/usr/local/lib/python3.6/ssl.py\", line 874, in read
    return self._sslobj.read(len, buffer)
  File \"/usr/local/lib/python3.6/ssl.py\", line 631, in read
    v = self._sslobj.read(len, buffer)
ConnectionResetError: [Errno 104] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File \"/usr/local/lib/python3.6/site-packages/requests/adapters.py\", line 449, in send
    timeout=timeout
  File \"/usr/local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 720, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File \"/usr/local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 400, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File \"/usr/local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 734, in reraise
    raise value. ... g of the above exception, another exception occurred:

Traceback (most recent call last):
  File \"./main.py\", line 11, in main
    application.run()
  File \"/home/google_bigquery_writer/app.py\", line 127, in run
    self.action_run()
  File \"/home/google_bigquery_writer/app.py\", line 179, in action_run
    incremental=incremental
  File \"/home/google_bigquery_writer/writer.py\", line 151, in write_table_sync
    incremental=incremental
  File \"/home/google_bigquery_writer/writer.py\", line 127, in write_table
    job_config=job_config
  File \"/usr/local/lib/python3.6/site-packages/google/cloud/bigquery/client.py\", line 1494, in load_table_from_file
    file_obj, job_resource, num_retries
  File \"/usr/local/lib/python3.6/site-packages/google/cloud/bigquery/client.py\", line 1806, in _do_resumable_upload
    response = upload.transmit_next_chunk(transport)
  File \"/usr/local/lib/python3.6/site-packages/google/resumable_media/requests/upload.py\", line 425, in transmit_next_chunk
    retry_strategy=self._retry_strategy,
  File \"/usr/local/lib/python3.6/site-packages/google/resumable_media/requests/_helpers.py\", line 136, in http_request
    return _helpers.wait_and_retry(func, RequestsMixin._get_status_code, retry_strategy)
  File \"/usr/local/lib/python3.6/site-packages/google/resumable_media/_helpers.py\", line 150, in wait_and_retry
    response = func()
  File \"/usr/local/lib/python3.6/site-packages/google/auth/transport/requests.py\", line 287, in request
    **kwargs
  File \"/usr/local/lib/python3.6/site-packages/requests/sessions.py\", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File \"/usr/local/lib/python3.6/site-packages/requests/sessions.py\", line 646, in send
    r = adapter.send(request, **kwargs)
  File \"/usr/local/lib/python3.6/site-packages/requests/adapters.py\", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
```
Úplně na konci je vidět, že to vyhazuje `requests.exceptions.ConnectionError` ne `BadRequest`. Dost možná je ten `requests.exceptions.ConnectionError` z úplně jiné knihovny (takže bude chybět ještě nějaký import), chtělo by to otestovat, že to skutečně zabere.